File: scribble_kiddies/java_codes/stringPermutations.java
package java_codes;

public class stringPermutations {
    public static void generate(String S, String ans) {
        if (S.length() == 0) {
            System.out.println(ans);
            return;
        }
        for (int i = 0; i < S.length(); i++) {
            generate(S.substring(0, i) + S.substring(i + 1), ans + S.charAt(i));
        }
    }

    public static void main(String args[]) {
        String S = "ABC";
        generate(S, "");
    }
}

File: scribble_kiddies/backTrack/stringPermutations.py
class stringPermutations:
    def generate(self,S,ans,i):
        if i == len(S):
            print(ans)
            return
        for j in range(i,len(S)):
            S[j],S[i] = S[i],S[j]
            self.generate(S,ans+S[i],i+1)
            S[j],S[i] = S[i],S[j]
A = stringPermutations()
A.generate(['A','B','C'],"",0)
